### PR TITLE
Jingle sip reINVITE

### PR DESCRIPTION
**Pull request by michalwski**
This PR extends Jingle/SIP integration translates:

1. From MongooseIM to SIP proxy. Jingle actions `source-add`, `source-remove`, `source-update` in established Jingle session to SIP's in-dialog INVITE message. A custom attribute `a=jingle-action:NAME` is added to the SDP packet to indicate what was the original action.
2. From SIP proxy to MongooseIM SIP's in-dialog INVITE (reINVITE) is translated to corresponding jingle action (based on the custom `a=jingle-action` attribute. If the attr is missing, the action is `transport-info`.

### CODE REVIEW

File: src/jingle_sip/jingle_sip_callbacks.erl [hunk 1]
```diff
@@ -99,23 +99,34 @@ translate_and_deliver_invite(Req, FromJID, FromBinary, ToJID, ToBinary) ->
...
     {CodecMap, SDP} = nksip_sdp_util:extract_codec_map(Body),
-    OtherEls = sip_to_jingle:parse_sdp_attributes(SDP#sdp.attributes),
+    RemainingAttrs = SDP#sdp.attributes,
+    OtherEls = sip_to_jingle:parse_sdp_attributes(RemainingAttrs),
 
     ContentEls = [sip_to_jingle:sdp_media_to_content_el(Media, CodecMap) || Media <- SDP#sdp.medias],
-    ?WARNING_MSG("ContentEls: ~p", [ContentEls]),
-    JingleEl = jingle_sip_helper:jingle_element(CallID, <<"transport-info">>, ContentEls ++ OtherEls),
+    Name = get_action_name_from_sdp(RemainingAttrs, <<"transport-info">>),
+    ?WARNING_MSG("name: ~p", [Name]),
```
Comment (arcusfelis) [inline src/jingle_sip/jingle_sip_callbacks.erl]: this WARNING_MSG is not needed.

File: big_tests/tests/jingle_sip_translator.erl [hunk 2]
```diff
@@ -291,6 +231,50 @@ in_invite_transaction_callback(Req, Socket, Tr, CorrespondingCall) ->
     ct:pal("in invite transaction callback:~n~p~n=========~n~p~n=========~n~p~n=========~n~p",
            [Req, Socket, Tr, CorrespondingCall]).
 
+dialog_callback(#sip{type = request, method = <<"ACK">>}, _, _) ->
+   ok;
+%% Below function forwards any in-dialog request to correspoding jingle session
+dialog_callback(#sip{type = request, hdrs = Hdrs} = Req, Socket, Tr) ->
+    ct:pal("dialog INVITE: ~p ~p", [Req, Tr]),
```
Comment (arcusfelis) [inline big_tests/tests/jingle_sip_translator.erl]: ct:log?

File: big_tests/tests/jingle_sip_translator.erl [hunk 3]
```diff
@@ -291,6 +231,50 @@ in_invite_transaction_callback(Req, Socket, Tr, CorrespondingCall) ->
...
+    Hdrs2 = [{via, [Via]}, {contact, [Contact]}, {'call-id', OtherCallID}, {from, From},
+                            {to, To} | Hdrs1],
+    ReqBack = Req#sip{uri = esip_codec:decode_uri(<<"sip:127.0.0.1:5600">>),
+                      hdrs = Hdrs2},
+    esip:request(Socket, ReqBack),
+    esip:make_response(Req, #sip{type = response,
+                                        status = 200
+                                        });
+dialog_callback(SIP, Socket, Tr) ->
+   ct:pal("uknown in-dialog sip msg: ~p", [SIP]).
```
Comment (arcusfelis) [inline big_tests/tests/jingle_sip_translator.erl]: typo

File: big_tests/tests/jingle_SUITE.erl [hunk 4]
```diff
@@ -294,6 +301,50 @@ jingle_session_is_intiated_and_canceled_by_receiver_on_different_node(Config) ->
...
+        accept_jingle_session(Alice, Bob, InviteStanza, InviteRequest),
+
+        %% then Alice sends source-remove
+        SourceRemoveStanza = escalus_stanza:to(jingle_source_add(InviteStanza), Bob),
+        escalus:send(Alice, SourceRemoveStanza),
+        SourceRemoveResult = escalus:wait_for_stanza(Alice),
+        escalus:assert(is_iq_result, [SourceRemoveStanza], SourceRemoveResult),
+        SourceRemoveToBob = escalus:wait_for_stanza(Bob),
+        assert_source_add_action(SourceRemoveToBob, InviteRequest)
+
```
Comment (arcusfelis) [inline big_tests/tests/jingle_SUITE.erl]: not needed new line

File: big_tests/tests/jingle_sip_translator.erl (hunk 3 above)
Comment (michalwski) [reply big_tests/tests/jingle_sip_translator.erl]: Where do you see the typo?

**Comment by michalwski** [comment]
@arcusfelis thanks for the review, I addressed most of the comments.

File: big_tests/tests/jingle_sip_translator.erl (hunk 3 above)
Comment (arcusfelis) [reply big_tests/tests/jingle_sip_translator.erl]: Should be "unknown"

**Pull request merged**